src/TriggerBinding/SqlTriggerTargetScaler.cs

```diff
@@ -14,24 +17,75 @@ internal sealed class SqlTriggerTargetScaler : ITargetScaler
+
+                // Instance concurrency value is set by the functions host when dynamic concurrency is enabled. See https://learn.microsoft.com/en-us/azure/azure-functions/functions-concurrency for more details.
+                int concurrency = context.InstanceConcurrency ?? this._maxChangesPerWorker;
 
-            // Instance concurrency value is set by the functions host when dynamic concurrency is enabled. See https://learn.microsoft.com/en-us/azure/azure-functions/functions-concurrency for more details.
-            int concurrency = context.InstanceConcurrency ?? this._maxChangesPerWorker;
+                return GetScaleResultInternal(concurrency, metrics.UnprocessedChangeCount);
+            }
+            catch (Exception ex)
+            {
+                // If the exception is SQL exception and indicates that the object name is invalid, it means that the global state and leases table are not created
+                // Check for the error number 208 or the error message https://learn.microsoft.com/en-us/sql/relational-databases/errors-events/mssqlserver-208-database-engine-error?view=sql-server-ver17
+                if (ex is SqlException sqlEx && sqlEx.Number == 208)
+                {
+                    // Check if we already tried to spin up the worker that starts the listener which will create those tables.
+                    // If we have already tried, we will return 0 as the target worker count.
+                    this._logger.LogWarning("Invalid object name detected. SQL trigger tables not found.");
+                    if (_lastTableCheck != DateTime.MinValue && DateTime.UtcNow - _lastTableCheck < TimeSpan.FromMinutes(2))
+                    {
+                        // If we have already checked within the last 2 minutes, we will return 0 as the target worker count.
+                        this._logger.LogWarning("Returning 0 as the target worker count since we have already checked for the SQL trigger tables within the last 2 minutes.");
+                        return new TargetScalerResult
+                        {
+                            TargetWorkerCount = 0
+                        };
+                    }
+                    else
+                    {
+                        // Check if there are any changes in the user table.
+                        int changes = await GetNumberOfChangesAsync(this._connectionString, this._userTable, this._logger, CancellationToken.None);
+                        // If there are changes in the user table, we need to spin up a worker to create the global state and leases table.
+                        if (changes > 0)
+                        {
+                            this._logger.LogWarning("There are changes in the change-tracking table for the user table, but the global state and leases table are not created. Spinning up a worker to create those tables.");
+                            _lastTableCheck = DateTime.UtcNow;
```

Comment:
We should only set this once, and consider renaming it something like `_firstTableCreationWarmupAttempt`